operator/controllers/testwithwatcher/kyma_controller_test.go

```diff
@@ -0,0 +1,149 @@
+package controllers_with_watcher_test
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/kyma-project/lifecycle-manager/operator/api/v1alpha1"
+	"github.com/kyma-project/lifecycle-manager/operator/controllers/testhelper"
+	"github.com/kyma-project/lifecycle-manager/operator/internal/deploy"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	admissionv1 "k8s.io/api/admissionregistration/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	namespace                      = "default"
+	webhookConfigNameTpl           = "%s-webhook"
+	servicePathTpl                 = "/validate/%s"
+	specSubresources               = "*"
+	statusSubresources             = "*/status"
+	expectedWebhookNamePartsLength = 4
+)
+
+var _ = Describe("Kyma with multiple module CRs in remote sync mode", Ordered, func() {
```

Comment:
I rewrite this test in https://github.com/kyma-project/lifecycle-manager/pull/294/files#diff-d449b17c7b4dc1b913e796f0eb376a9aab8994bee23bc46857fc3c812130b0baR35-R62